### PR TITLE
⭐  New Source: Goldcast 

### PR DESCRIPTION
**Pull request by FVidalCarneiro**
## What

This Pull Request aims in responding to issue (https://github.com/airbytehq/airbyte/discussions/38785) by proposing a first version of the Goldcast source connector.

## How

This PR adds 8 base streams of the Goldcast API:

- organizations
- events
- webinars
- event_members
- event_agenda
- agenda_items
- discussion_groups
- tracks

This connector was developed using the [low-code development framework](https://docs.airbyte.com/connector-development/config-based/low-code-cdk-overview).

Schemas of the ingested objects are defined in the `schemas/` directory.

## Review guide

1. manifest.yaml
2. README.md
3. No particular order after

## User Impact

User will have access to 8 streams of a new stream.

Attention: It is worth mentioning that due to field limitations on Goldcast side, only full append stream ingestion modes are supported. This is no cursor field available yet for these API objects.

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [x] NO ❌ (as soon as it is merged, customers could potentially use it in production)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207702493726294

### CODE REVIEW

File: airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json:4]: This should always be an object instead of array.
Add `additionalProperties: true`

File: airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json:33]: Removed this as can make hard to keep compatibility with future releases

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left a comment need to be replicated to all schemas. Missing documentation for the connector too. Let me know if you need any help to solve them.

File: airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json:4]: This is more an internal requirement to make schema more flexible for future updates in the API.

File: airbyte-integrations/connectors/source-goldcast-api/metadata.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast-api/metadata.yaml:18]: Can you bump this to `1.2.2` please? We just released it last week, but it's already used in a bunch of connectors if you want to find the SHA

File: airbyte-integrations/connectors/source-goldcast-api/metadata.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast-api/metadata.yaml:27]:
```suggestion
  name: Goldcast
```

It would be rather obvious that it's an API ;)

File: airbyte-integrations/connectors/source-goldcast-api/metadata.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast-api/metadata.yaml:31]:
```suggestion
  documentationUrl: https://docs.airbyte.com/integrations/sources/goldcast
```

File: airbyte-integrations/connectors/source-goldcast-api/acceptance-test-config.yml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast-api/acceptance-test-config.yml:7]:
```suggestion
      - spec_path: "source_goldcast/spec.yaml"
```

@FVidalCarneiro If you do end up using the `goldcast` for the name, make sure to rename `goldcast_api` and `goldcast-api` everywhere, including the folder name

File: airbyte-integrations/connectors/source-goldcast-api/main.py
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast-api/main.py:9]:
```suggestion
from source_goldcast import SourceGoldcast
```

File: airbyte-integrations/connectors/source-goldcast-api/integration_tests/abnormal_state.json
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast-api/integration_tests/abnormal_state.json:1]: Remove this file if you're not doing integration tests

**Review by natikgadzhi — COMMENTED** [review]
Thank you for contributing a new connector, @FVidalCarneiro! 

I think we should seriously upgrade our low-code connector template, a lot of things here seem unnecessary. 

A few things to wrap up for this to be merged:
- [ ] Let's rename to just `goldcast`. It could be easier to regenerate from template and paste the `manifest.yaml` file in the new connector.
- [ ] Remove unused integration test files. Search for `TODO` and either implement or remove them.
- [ ] Upgrade to base_image 1.2.2

File: airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json
Comment (FVidalCarneiro) [reply airbyte-integrations/connectors/source-goldcast-api/source_goldcast_api/schemas/agenda_items.json:4]: @marcosmarxm , I agree, I am referring to the comment on the array vs object. I think the API response is that of an array, not an object

**Comment by FVidalCarneiro** [comment]
Hi @natikgadzhi, I believed I replied to all your feedback on this PR. Just need to understand how to populate the documentation page, can you give me some information on that ?

**Comment by natikgadzhi** [comment]
Okay, this is great, but I need two things: 
1. Let's inline schemas into the manifest file. @btkcodedev how do you usually do that? Please don't say manually. 
2. We need to add a documentation file in `docs/` for `goldcast`. Follow whatever structure the other files have.

**Comment by btkcodedev** [comment]
> Okay, this is great, but I need two things:
> 
> 1. Let's inline schemas into the manifest file. @btkcodedev how do you usually do that? Please don't say manually.
> 2. We need to add a documentation file in `docs/` for `goldcast`. Follow whatever structure the other files have.

Oh, that's with the help of a magical command from `airbyte-ci`
For this PR it would be 
`airbyte-ci connectors --name source-goldcast migrate_to_inline_schemas`

It will automatically inline schemas and delete the existing JSON schemas, It Works like charm🎉

**Comment by FVidalCarneiro** [comment]
Hello @natikgadzhi ,

1. I have run the inline schema command as suggested by @btkcodedev - the command is `migrate-to-inline_schemas` ;) It ran fine and I think the manifest.yml file now has the expected structure. However, I did notice the script did not finishing running for me, I had to keyboard interrupt it as it looked blocked and had already produced the desired output.
2. Regarding the second documentation point, I am not sure how to contribute to the official airbyte documentation pages. Happy to write them ofc, just tell me how to, is there a specific procedure to follow ? 

Let me know what the next steps are for you.

Thanks !

**Comment by btkcodedev** [comment]
Hi, @FVidalCarneiro, 
1. Nice to know the migrate command worked, the finishing issue might be due to the prettier compatibility issue, as the final step was to format the manifest. It works if you have npx and prettier installed
It runs `npx prettier --write airbyte-integrations/connectors/source-goldcast/source_goldcast/manifest.yaml` :smile: 

2. For writing documentation, please goto `~/airbyte/docs/integrations/sources` folder and create a new file.md named your connector (goldcast.md)
For more insight into what to write, please look a other connectors for reference, the header should always named with TitleCase as your connector name `# Goldcast` and there should be a version update with the date and PR# just as other connectors

**Comment by FVidalCarneiro** [comment]
@btkcodedev , @marcosmarxm , just added a v1 documentation

**Comment by natikgadzhi** [comment]
Wow that's weird, CI just decided to fail everything with no explanation. Restarting ;)

**Comment by FVidalCarneiro** [comment]
HI @natikgadzhi , I cant seem to access the failed CI job logs. Could you let me know what I need to change ? 
Failing connection changes I guess are normal as you need to generate a Goldcast access token secret on your side for that to work

**Comment by natikgadzhi** [comment]
@FVidalCarneiro I know you're going to sort of hate me, but I have a nice idea.

I'm assuming that this whole connector is fully low-code, and the only custom stuff it has is integration tests configured catalogs. If so, let's delete the bunch of boilerplate, and try our new connector build packaging: 

1. Move `manifest.yaml` into the root of the connector directory.
2. Delete all python files, including `__init__.py`, `run.py` and `main.py` and friends. 
3. Keep `readme.md`, `manifest.yaml`, `metadata.yaml`, `integration_tests/*.json` and `acceptance-test-configuration.yaml`. 
4. Once https://github.com/airbytehq/airbyte/pull/39906 ships, and your connector looks something like this https://github.com/airbytehq/airbyte/pull/39907, it'll be super easy to merge and maintain.

Our timeline to merge the build system is probably next week.

**Comment by natikgadzhi** [comment]
Getting close: 

```
source-goldcast - ❌ Failed - Connectors must have valid metadata.yaml file: Metadata file is invalid: Validation error: 1 validation error for ConnectorMetadataDefinitionV0
data -> releaseDate
  invalid date format (type=value_error.date).
source-goldcast - ❌ Failed - Connectors must have an icon: Icon file is missing. Please create an icon file at the root of the connector code directory..
Error: 2 checks failed
```

**Comment by natikgadzhi** [comment]
It's OK to not do slim buidl, but at least we should fix metadata. I'm very happy to merge this once the build succeeds

File: airbyte-integrations/connectors/source-goldcast/source_goldcast/source.py
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast/source_goldcast/source.py:16]: In theory you can drop this file and just inherit from YamlDeclarativeSource in `main.py`, but it's okay either way

File: airbyte-integrations/connectors/source-goldcast/metadata.yaml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast/metadata.yaml:9]:
```suggestion
      enabled: true
```

File: airbyte-integrations/connectors/source-goldcast/source_goldcast/schemas/event_agenda.json
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast/source_goldcast/schemas/event_agenda.json:1]: I would prefer to inline schema in the yaml manifest, but again, can come later. Esp if you're interested in making more sources!

File: airbyte-integrations/connectors/source-goldcast/pyproject.toml
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-goldcast/pyproject.toml:19]: Can you please bump this to "< 3" and run `poetry lock`? I'd love to start this on the fresh CDK. I will merge if test pass without it, but please follow up a PR to upgrade.

**Review by natikgadzhi — APPROVED** [review]
Overall, this looks great and I'd be happy to merge this once CI passes. This should then be available promptly once it populates to registry. 

I left some suggestions, and I'd love you to act on them if you have time.

**Comment by natikgadzhi** [comment]
If you can allow me to make changes to this PR, I'm happy to make some changes myself.

File: airbyte-integrations/connectors/source-goldcast/source_goldcast/schemas/event_agenda.json
Comment (FVidalCarneiro) [reply airbyte-integrations/connectors/source-goldcast/source_goldcast/schemas/event_agenda.json:1]: @natikgadzhi , Its done as well ! Would you like me to delete these json schemas ? Is it not required ?

File: airbyte-integrations/connectors/source-goldcast/pyproject.toml
Comment (FVidalCarneiro) [reply airbyte-integrations/connectors/source-goldcast/pyproject.toml]: Corrected ! @natikgadzhi

**Comment by FVidalCarneiro** [comment]
> If you can allow me to make changes to this PR, I'm happy to make some changes myself.

Regarding this point @natikgadzhi , I would gladly share you edit access but I had this problem in another PR where I tried to follow (https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests) and think it is not possible. Here is the link to the git discussion (https://github.com/airbytehq/airbyte/pull/34166#issuecomment-1935589128).

**Comment by natikgadzhi** [comment]
Merging. This should show up on the catalog in an hour or so

**Comment by FVidalCarneiro** [comment]
@natikgadzhi , thank you for your support with this. Happy to have our contributions integrated 😄

**Pull request merged**
